### PR TITLE
Add a SQL database backed data store

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: gradle/check-licenses.gradle [hunk 1]
```diff
@@ -15,6 +15,7 @@
  * The lists of the license we accept.
  */
 ext.acceptedLicenses = [
+  'Apache v2',
```
Comment (cleishm) [inline gradle/check-licenses.gradle]: Put this in the aliases section instead?

File: kv/src/main/kotlin/net/consensys/cava/kv/SQLKeyValueStore.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  init {
+    val config = BoneCPConfig()
+    config.jdbcUrl = jdbcurl
+
+    connectionPool = BoneCP(config)
+  }
+
+  override suspend fun get(key: Bytes): Bytes? = withContext(dispatcher) {
+    connectionPool.connection.use {
```
Comment (cleishm) [inline kv/src/main/kotlin/net/consensys/cava/kv/SQLKeyValueStore.kt]: According to https://jolbox.com/features.html, obtaining a connection can be done async. But I haven't seen where in the javadoc 🤔

File: gradle/check-licenses.gradle (hunk 1 above)
Comment (atoulme) [reply gradle/check-licenses.gradle]: Missed that. Fixed!

File: kv/src/main/kotlin/net/consensys/cava/kv/SQLKeyValueStore.kt (hunk 2 above)
Comment (atoulme) [reply kv/src/main/kotlin/net/consensys/cava/kv/SQLKeyValueStore.kt]: OK that's changed to use `asyncConnection`, and using coroutines guava integration to use the right await() implementation.

**Pull request merged**